### PR TITLE
RPP: Add autoloader

### LINKED ISSUE #6677: Payment Process: Add PSR-4 autoloader
### Description

Add PSR-4 autoloading to the new `src` directory, created in https://github.com/Automattic/woocommerce-payments/issues/6675.

### Acceptance criteria

- No additional `include` or `require` statements for loading the `src` directory.
- The PHPCS rule regarding WordPress file names is disabled for the directory, being replaced by another one ([example](https://github.com/suin/phpcs-psr4-sniff)) for PSR-4.
- The Composer autoloader is used.
- If needed, add a new `npm` command for generating class maps.

### Testing instructions

We should be able to instantiate the payment service anywhere (ex. `woocommerce-payments.php`) without needing to include the file.

### Additional context

Epic: https://github.com/Automattic/woocommerce-payments/issues/6673
Project: paJDYF-9hL-p2

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
Fixes #6677.

#### Changes proposed in this Pull Request

Use Composer's PSR-4 autoloader for the `src` directory.

#### Testing instructions

Load any page after appending `?load` to the URL, ex. `http://localhost:8082/?load`. This snippet is just there for testing, and will be removed.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests: Nothing to test.
- [x] Tested on mobile (does not apply)

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6856` or branch name `rpp/6677-psr-autoloader` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=rpp/6677-psr-autoloader) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: f771ddab21aa64ad79afe2d8befc4044f2fafd7a
- Build time: 2023-08-01 15:37:32 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/rpp-6677-psr-autoloader
Comment (htdat) [inline changelog/rpp-6677-psr-autoloader]: Nitpick: 

```suggestion
Add PSR-4 autoloading for the src directory.
```

File: includes/class-wc-payments.php
Comment (htdat) [inline includes/class-wc-payments.php]: May setup a new PHPUnit test file. That would simply validate `PaymentService::class` to its full qualified class name. 

Also use the namespace for it, something like these in Woo core: 
- https://github.com/woocommerce/woocommerce/blob/079274d4284be51642c4b08d9c57827eff4e5596/plugins/woocommerce/composer.json#L67-L69
- https://github.com/woocommerce/woocommerce/blob/079274d4284be51642c4b08d9c57827eff4e5596/plugins/woocommerce/tests/php/src/Internal/Features/FeaturesControllerTest.php#L16-L17

**Review by htdat — COMMENTED** [review]
Initial feedback.

File: includes/class-wc-payments.php
Comment (RadoslavGeorgiev) [reply includes/class-wc-payments.php]: > May setup a new PHPUnit test file. That would simply validate PaymentService::class to its full qualified class name.

Good idea, done now.

> Also use the namespace for it, something like these in Woo core.

I did use a namespace for the test, as was my plan for when I had to write another one. I did not see a need to add it to the autoloader though, it seems to just work, as with other PHP files during unit tests 🤔

Comment (htdat) [reply includes/class-wc-payments.php]: >  I did not see a need to add it to the autoloader though, it seems to just work, as with other PHP files during unit tests 

I think that's because PHPUnit will load all `.php` files based on [its configuration](https://github.com/Automattic/woocommerce-payments/blob/9da528c0dd6efef2c0fabce99123c97617acb1fb/phpunit.xml.dist#L11-L12). 

If we also config the autoload for these test files, suddenly these test files will be available outside of test runners too. That's unnecessary IMO.

**Review by htdat — APPROVED** [review]
LGTM! We can merge after #6805 landing into `develop` too.

**Pull request merged**
